addons/account/models/account_move.py

```diff
@@ -2980,9 +2980,10 @@ def is_write_needed(line, values):
                 and (
                     field_has_changed(moves_values_before, move, 'currency_id')
                     or field_has_changed(moves_values_before, move, 'move_type')
+                    or field_has_changed(moves_values_before, move, 'invoice_currency_rate') and move.company_id.tax_calculation_rounding_method == 'round_globally'
```

Comment:
@gteboul Awaiting the explanation about why this specific condition about round_globally is necessary ~~